### PR TITLE
gh-91063: Update macOS installer to SQLite 3.38.4.

### LINKED ISSUE #91063: Update Windows and MacOS installer to SQLite 3.38.4
BPO | [46907](https://bugs.python.org/issue46907)
--- | :---
Nosy | @pfmoore, @ronaldoussoren, @tjguk, @ned-deily, @zware, @zooba, @tirkarthi, @erlend-aasland, @felixxm, @180909
PRs | <li>python/cpython#31655</li><li>python/cpython#31656</li><li>python/cpython#32147</li><li>python/cpython#32148</li><li>python/cpython#32149</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2022-03-03.08:27:32.906>
labels = ['OS-mac', 'type-feature', 'OS-windows', '3.11']
title = 'Update Windows and MacOS installer to SQLite 3.38.2'
updated_at = <Date 2022-03-31.18:05:03.473>
user = 'https://github.com/felixxm'
```

bugs.python.org fields:
```python
activity = <Date 2022-03-31.18:05:03.473>
actor = 'erlendaasland'
assignee = 'none'
closed = False
closed_date = None
closer = None
components = ['macOS', 'Windows']
creation = <Date 2022-03-03.08:27:32.906>
creator = 'felixxm'
dependencies = []
files = []
hgrepos = []
issue_num = 46907
keywords = ['patch']
message_count = 17.0
messages = ['414412', '414413', '414414', '415015', '415251', '415265', '415266', '415269', '415270', '415271', '416140', '416274', '416429', '416442', '416446', '416450', '416453']
nosy_count = 10.0
nosy_names = ['paul.moore', 'ronaldoussoren', 'tim.golden', 'ned.deily', 'zach.ware', 'steve.dower', 'xtreak', 'erlendaasland', 'felixxm', 'wangjiahua']
pr_nums = ['31655', '31656', '32147', '32148', '32149']
priority = 'normal'
resolution = None
stage = 'patch review'
status = 'open'
superseder = None
type = 'enhancement'
url = 'https://bugs.python.org/issue46907'
versions = ['Python 3.11']
```

</p></details>

### PR DESCRIPTION
**Pull request by felixxm**
https://bugs.python.org/issue46907

https://bugs.python.org/issue46907

#91063

### CODE REVIEW

**Review by erlend-aasland — APPROVED** [review]

**Comment by erlend-aasland** [comment]
@ned-deily, would you mind merging this?

**Comment by erlend-aasland** [comment]
btw, @felixxm, you may need to pull in `main`[^1]  on both PRs in order to retrigger the CLA check.

[^1]: `git pull --no-ff main`

**Comment by ned-deily** [comment]
> @ned-deily, would you mind merging this?

Thanks, I will be merging it soon along with other installer updates for b1.

**Comment by felixxm** [comment]
> btw, @felixxm, you may need to pull in `main`[1](#user-content-fn-1-bcb6c5a54a9a1b31034def13328130a6) on both PRs in order to retrigger the CLA check.

Rebased.

**Comment by felixxm** [comment]
Updated to SQLite 3.38.3.

**Review by erlend-aasland — APPROVED** [review]

File: Mac/BuildScript/build-installer.py [hunk 1]
```diff
@@ -358,14 +358,12 @@ def library_recipes():
                   ),
           ),
           dict(
-              name="SQLite 3.38.1",
-              url="https://sqlite.org/2022/sqlite-autoconf-3380100.tar.gz",
-              checksum="5af57892dc0993af596bef56931db23f",
+              name="SQLite 3.38.3",
+              url="https://sqlite.org/2022/sqlite-autoconf-3380300.tar.gz",
```
Comment (ned-deily) [inline Mac/BuildScript/build-installer.py]: checksum= missing?

Comment (erlend-aasland) [reply Mac/BuildScript/build-installer.py]: Good catch!

Comment (erlend-aasland) [reply Mac/BuildScript/build-installer.py]: cc. @felixxm

Comment (felixxm) [reply Mac/BuildScript/build-installer.py]: Sorry, fixed.

**Review by erlend-aasland — APPROVED** [review]

**Comment by ned-deily** [comment]
[3.38.4](https://sqlite.org/2022/sqlite-autoconf-3380400.tar.gz) was released today.

**Comment by felixxm** [comment]
> [3.38.4](https://sqlite.org/2022/sqlite-autoconf-3380400.tar.gz) was released today.

Great timing :smile: . Updated.

**Review by erlend-aasland — APPROVED** [review]

**Pull request merged**